Add ThreadChannel interface

RealSeek opened the pull request:
- introduce the ThreadChannel interface to represent KOOK's thread channels (帖子频道)
- this interface extends NonCategoryChannel and provides methods to get and set the channel's topic
- includes comprehensive Javadoc explaining features, API limitations, and usage examples

RealSeek committed:
✨ feat(channel): add ThreadChannel interface

- introduce the ThreadChannel interface to represent KOOK's thread channels (帖子频道)
- this interface extends NonCategoryChannel and provides methods to get and set the channel's topic
- includes comprehensive Javadoc explaining features, API limitations, and usage examples

RealSeek committed:
✨ feat(thread): add entities and methods for thread channels

This commit introduces comprehensive support for KOOK's thread channel feature.

【feat】 Add `Thread` and `ThreadPost` interfaces to model threads and their replies, providing
access to essential data like ID, author, content, and timestamps.

【feat】 Extend `ThreadChannel` with functionality to:
- Create new threads (`createThread`).
- Fetch individual threads (`getThread`) and lists of threads (`getThreads`).
- List thread categories (`getCategories`) via the new `ThreadCategory` interface.

【docs】 Update `ThreadChannel` Javadoc with usage examples for the new features.

RealSeek committed:
✨ feat(thread)!: overhaul thread API for full functionality

- 【feat】Introduced comprehensive properties for `Thread` and `ThreadPost` entities to align with
  the full capabilities of the KOOK API, including status, timestamps, and content metadata.
- 【feat】Added detailed permission management to `ThreadChannel.ThreadCategory` with methods for
  default permissions (`getAllow`, `getDeny`) and specific overrides (`getRoles`).
- 【feat】Added new `CategoryPermissionOverride` interface to model role/user-specific permissions.
- 【refactor】Changed the return type of `getContent()` on `Thread` and `ThreadPost` from `String`
  to `MultipleCardComponent` to support rich card-based content.
- 【refactor】Renamed `ThreadPost.getParentPostId()` to `getReplyToPostId()` for clarity.

BREAKING CHANGE: The `getContent()` method in `Thread` and `ThreadPost` now returns a
`MultipleCardComponent` object instead of a raw `String`.

BREAKING CHANGE: The `ThreadPost.getParentPostId()` method has been renamed to
`getReplyToPostId()`.

BREAKING CHANGE: The `Thread.isPinned()` method has been removed.

huanmeng-qwq commented on src/main/java/snw/jkook/entity/channel/ThreadChannel.java:
```diff
@@ -0,0 +1,220 @@
+        int getAllow();
+    }
+
+}
```
考虑使用`RolePermissionOverwrite`/`UserPermissionOverwrite`代替，将其作为通用的接口

huanmeng-qwq commented on src/main/java/snw/jkook/entity/channel/ThreadChannel.java:
```diff
@@ -0,0 +1,220 @@
+     * @param topic The new topic content
+     * @throws IllegalArgumentException if the topic is null or exceeds maximum length
+     */
+    void setTopic(String topic);
```
`topic` 在JKook项目中仅在 `TextChannel` 存在，需要确认ThreadChannel是否存在该字段？

huanmeng-qwq commented on src/main/java/snw/jkook/entity/thread/Thread.java:
```diff
@@ -0,0 +1,211 @@
+     * <p>Status values:
+     * <ul>
+     *   <li>0 - Normal</li>
+     *   <li>1 - Deleted</li>
```
1代表审核中，2代表审核通过, 3代表编辑审核中

huanmeng-qwq commented on src/main/java/snw/jkook/entity/thread/Thread.java:
```diff
@@ -0,0 +1,211 @@
+     *
+     * @return The creation timestamp in milliseconds
+     */
+    long getTimeStamp();
```
```suggestion
    long getCreateTime();
```

huanmeng-qwq commented on src/main/java/snw/jkook/entity/thread/Thread.java:
```diff
@@ -0,0 +1,211 @@
+     * <ul>
+     *   <li>0 - Not deleted</li>
+     *   <li>1 - Deleted by author</li>
+     *   <li>2 - Deleted by moderator</li>
```
3: 审核删除

RealSeek committed:
♻️ refactor!(thread): align thread and thread channel APIs with kook platform

This commit updates the Thread and ThreadChannel interfaces to match recent
changes in the Kook platform API, introducing new statuses and refactoring
deprecated or inconsistent elements.

【feat】
- Add new thread status codes: Under review, Approved, and Edit under review
- Add new thread deletion type: Deleted by review

【refactor】
- Rename `Thread#getTimeStamp` to `Thread#getCreateTime` for clarity
- Remove `getTopic()` and `setTopic()` from `ThreadChannel`
- Replace `CategoryPermissionOverride` with the generic `Channel.PermissionOverwrite`

BREAKING CHANGE: Several methods and types related to threads have been changed or removed.
- The method `Thread#getTimeStamp()` has been renamed to `Thread#getCreateTime()`.
- The methods `getTopic()` and `setTopic()` have been removed from the `ThreadChannel` interface.
- The nested interface `ThreadChannel.CategoryPermissionOverride` has been removed.
- The return type of `ThreadChannel.Category#getRoles()` is now `Collection<Channel.PermissionOverwrite<?>>`.

SNWCreations commented on src/main/java/snw/jkook/entity/thread/Thread.java:
避免使用与 Java 标准库中已存在的类的名称

RealSeek committed:
♻️ refactor(thread)!: clarify distinction between thread posts and replies

- The `Thread` interface has been removed and its role is now filled by `ThreadPost`.
- The `ThreadPost` interface now represents the main post/topic in a thread channel.
- A new `ThreadReply` interface has been introduced to represent replies to a `ThreadPost`.
- This refactoring provides a clearer and more intuitive API for handling thread entities.

BREAKING CHANGE: The `Thread` and `ThreadPost` interfaces have been restructured.
The `snw.jkook.entity.thread.Thread` interface is removed. Code should be updated to use
`ThreadPost` for main thread entries and the new `ThreadReply` interface for replies.
Methods in `ThreadChannel` have been updated to return `ThreadPost` instead of `Thread`.

SNWCreations commented on src/main/java/snw/jkook/entity/channel/ThreadChannel.java:
```diff
@@ -0,0 +1,165 @@
+     * @return The created thread post
+     * @throws IllegalArgumentException if title or content is null or empty
+     */
+    ThreadPost createThread(String title, String content, @Nullable String categoryId);
```
rename to createThreadPost

SNWCreations commented on src/main/java/snw/jkook/entity/channel/ThreadChannel.java:
```diff
@@ -0,0 +1,165 @@
+     * @return The thread, or null if not found
+     */
+    @Nullable
+    ThreadPost getThread(String threadId);
```
rename to getThreadPost

SNWCreations commented on src/main/java/snw/jkook/entity/channel/ThreadChannel.java:
```diff
@@ -0,0 +1,165 @@
+     * @param categoryId The category ID to filter threads, null for all threads
+     * @return A page iterator for threads
+     */
+    PageIterator<Collection<ThreadPost>> getThreads(@Nullable String categoryId);
```
rename to getThreadPosts

RealSeek committed:
♻️ refactor(channel)!: rename thread methods for clarity

- rename `getThread` to `getThreadPost` to avoid ambiguity and improve clarity
- rename `getThreads` to `getThreadPosts` for consistency with the singular method name

BREAKING CHANGE: The methods `getThread(String)` and `getThreads(String)` in the `ThreadChannel`
interface have been renamed to `getThreadPost(String)` and `getThreadPosts(String)` respectively.

SNWCreations commented on src/main/java/snw/jkook/entity/thread/ThreadReply.java:
```diff
@@ -0,0 +1,132 @@
+     * Delete this reply.
+     *
+     * <p><b>Note:</b> This action requires appropriate permissions.
+     */
```
使用 RequirePermission 注解标明此方法需要的权限，如果需要的权限未定义，则在 Permission 枚举增加定义。
